proposals/left-right-join-in-query-expressions.md

```diff
@@ -0,0 +1,235 @@
+
+```c#
+from student in students
+left join department in departments on student.DepartmentID equals department.ID
+select new { student.Name, department?.Name }
+```
+
+## Motivation
+[motivation]: #motivation
+
+LINQ has a [Join operator](https://learn.microsoft.com/dotnet/api/system.linq.enumerable.join), which, like its SQL INNER JOIN counterpart, correlates elements of two sequences based on matching keys; the C# language has a corresponding `join` clause which translates to this operator ([section 12.20.3.5 of the C# specs](https://learn.microsoft.com/en-us/dotnet/csharp/language-reference/language-specification/expressions#122035-from-let-where-join-and-orderby-clauses)).
+
+In addition to INNER JOIN, SQL also has LEFT JOIN, which returns outer elements even if there's no corresponding inner ones; LINQ and C#, in contrast, lack this operator. [The LINQ conceptual documentation shows how to combine existing operators to achieve a left join](https://learn.microsoft.com/en-us/dotnet/csharp/linq/standard-query-operators/join-operations#perform-left-outer-joins):
+
+```c#
+var query =
+    from student in students
+    join department in departments on student.DepartmentID equals department.ID into gj
+    from subgroup in gj.DefaultIfEmpty()
+    select new
+    {
+        student.FirstName,
+        student.LastName,
+        Department = subgroup?.Name ?? string.Empty
+    };
+```
+
+Or using method syntax:
+
+```c#
+var query = students
+    .GroupJoin(departments, student => student.DepartmentID, department => department.ID, (student, departmentList) => new { student, subgroup = departmentList })
+    .SelectMany(
+        joinedSet => joinedSet.subgroup.DefaultIfEmpty(),
+        (student, department) => new
+        {
+            student.student.FirstName,
+            student.student.LastName,
+            Department = department.Name
+        });
+```
+
+Although functionality sufficient for expressing a left join operation, this combining approach has the following drawbacks:
+
+- It's complicated, requiring combining multiple different LINQ operators in a specific way to form a complex construct, and is easy to accidentally get wrong. Many EF users have complained about the complexity of this construct for expressing a simple SQL LEFT JOIN.
+- It is inefficient - the combination of operators adds significant overhead compared to a single operator using an internal lookup table (or "hash join", as Join is implemented).
+
+https://github.com/dotnet/runtime/issues/110292 is an approved API proposal for introducing LeftJoin and RightJoin operators into the System.Linq for .NET 10; see that proposal for additional performance information and benchmarks. Aside from allowing users to more easily express SQL LEFT/RIGHT JOIN when using a LINQ provider (such as EF Core), it would also allow both easier and more efficient in-memory left/right joins, exactly in the way that inner joins are already supported.
```

Comment:
Oh.  you added that here.  Could you also show the API shape for these methods here, alongside the existing join api methods?